align_slice.py

```diff
@@ -28,3 +28,27 @@ def align_and_slice(s, dimension='t2', convwidth=500, threshold=0.05, fl=None):
     if fl is not None:
         fl.pop_marker()
     return s[dimension:slices]
+def correlation_align(s,avg,convwidth=0,dimension='t2',fl=None):
+    assert not s.get_ft_prop(dimension), 'I want time-domain data'
+    assert not avg.get_ft_prop(dimension), 'I want time-domain data'
+    avg.run(conj)
+    R = pi*convwidth # Lorentzian FWHM formula -- R = 1/T2, FWHM = R/pi
+    # ALSO convolve both
+    check = avg * s * exp(-2*R*abs(s.fromaxis(dimension)))
+    assert s.get_ft_prop(dimension,['start','freq']) is not None, "should be FT'd first, so the startpoint is set"
+    check.ft('t2', pad=2**14)
+    check.reorder('t2',first=False)
+    forplot = abs(check)['t2':(-50,50)]
+    myline = forplot.C.argmax('t2')
+    forplot.smoosh(['ph1','ph2','power'],'indirect'
```

Comment:
so because one of these dimensions is power I am assuming this funciton is only applicable to ODNP experiments?

Reply:
actually, no, that's not true ... I will edit this in response